src/libraries/System.Linq/src/System/Linq/OrderedEnumerable.cs

```diff
@@ -158,6 +173,16 @@ internal override CachingComparer<TElement> GetComparer(CachingComparer<TElement
                 : new CachingComparerWithChild<TElement, TKey>(_keySelector, _comparer, _descending, childComparer);
             return _parent != null ? _parent.GetComparer(cmp) : cmp;
         }
+
+        protected override bool CanInPlaceSort()
+        {
+            if (_parent != null)
+            {
+                return false;
+            }
+            Type elementType = typeof(TElement);
+            return elementType.IsValueType || (elementType.IsClass && !elementType.IsArray);
```

Comment:
This expression could be inlined by the JIT if you avoided assigning `typeof(TElement)` to an intermediate variable:

https://sharplab.io/#v2:C4LghgzgtgPgAgJgIwFgBQcAMACAygCzACcAHAGTACMA6AJQFcA7YASygFMBuddAbQCkWwAOLtG7IiwDGACmABPEuwD2AMxktmASi0BddHADM2RNgDCAHgAqAUQA27DswB82AN7psX7J+8lJAG5gwOzYypQAVuxSwNgA+iTEYrEAvNiM9HZ23GjePrnevl5G2JTKynbmYIwAkowACnZgUuy4ykTAMlpF7j15LKrYMglJzNgAhGkZWd0FeV4ec/N5cADs2KpgdhBcfd4AvjxLeVaKoewOTsCnSthpCkpqMraXyVo5y8XrF47JN+zUGoQABqW3o7H+2BgMCGPyu/0BEDMTQgEGwADJ0RM4X8zoiAIJEIhgeTvHqHJY9EplCpVWoNJotNodBBdHqLT7YAZDEZEZITKaZOyzTkczneNYbLY7D6fCmcvZfbAPFTqF6/bSI0F2cGQ6FDFVPdVXLSI5GQNGY5VnI32DXAU1AwnE0myrwU9D7IA=